PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/function/runner/localrun/FunctionRunConfiguration.java

```diff
@@ -221,16 +236,37 @@ public void initializeDefaults(Module module) {
         if (StringUtils.isEmpty(this.getHostJsonPath())) {
             this.setHostJsonPath(Paths.get(getProject().getBasePath(), "host.json").toString());
         }
-
-        if (StringUtils.isEmpty(this.getLocalSettingsJsonPath())) {
-            this.setLocalSettingsJsonPath(Paths.get(getProject().getBasePath(), "local.settings.json").toString());
+        final String localSettingsJsonPath = this.getLocalSettingsJsonPath(module);
+        if (StringUtils.isNotEmpty(localSettingsJsonPath)) {
+            this.setLocalSettingsJsonPath(localSettingsJsonPath);
         }
         try {
-            final Map<String, String> localSettings = FunctionAppSettingsTableUtils.getAppSettingsFromLocalSettingsJson(new File(this.getLocalSettingsJsonPath()));
+            final Map<String, String> localSettings = FunctionAppSettingsTableUtils.getAppSettingsFromLocalSettingsJson(new File(localSettingsJsonPath));
             FunctionUtils.saveAppSettingsToSecurityStorage(getAppSettingsKey(), localSettings);
         } catch (final Throwable throwable) {
             // swallow exception when load app settings
         }
+        try {
+            prepareBeforeRunTasks();
+        } catch (final Throwable throwable) {
+            AzureMessager.getMessager().warning("Failed to add Azure resource connections, please add it manually in run configuration if necessary");
+        }
+    }
+
+    // workaround to correct before run tasks in quick launch as BeforeRunTaskAdder may not work or have wrong config in task in this case
+    private void prepareBeforeRunTasks() {
```

Comment:
may try save module in craeteConfigurationFromContext in `RunConfigurationProducer`